File: samples/TwoSteps/coins/public/java/coins.java

public class coins {
	
	int[] coin_flips(int[] b, int c) {
		int[] flips = new int[1];
		if (b[c] == 1) {
			flips[0] = 0;
		} else {
			flips[0] = 4;
		}
		return flips;
	}
	
	int find_coin(int[] b) {
		if (b[0] == 0) {
			return 0;
		}
		return 7;
	}
}

File: samples/TwoSteps/coins/public/pas/coins.pas
unit coins;

interface
type
    TIntArray = array of longint;

function coin_flips(b: TIntArray; c: longint): TIntArray;
function find_coin(b: TIntArray): longint;

implementation

function coin_flips(b: TIntArray; c: longint): TIntArray;
var
    flips: TIntArray;
begin
    setLength(flips, 1);
    if b[c] = 1 then
        flips[0] := 0
    else
        flips[0] := 4;
	coin_flips := flips;
end;

function find_coin(b: TIntArray): longint;
begin
    if b[0] = 0 then
    begin
        find_coin := 0;
        exit;
    end;
    find_coin := 7;
end;

end.
